dali/dfu/dfuutil.cpp

```diff
@@ -857,21 +857,61 @@ class CDFUhelper: public CInterface, implements IDFUhelper
             else
                 throwError1(DFUERR_DFileNotFound, lfn);
         }
-        else {
-            Owned<IFileDescriptor> fdesc = file->getFileDescriptor();
-            Owned<IPropertyTree> t = fdesc->getFileTree();
+        else
+        {
+            Owned<IPropertyTree> t = queryDistributedFileDirectory().getFileTree(lfn, user);
             toXML(t, out, true);
         }
         return out;
     }
 
-    void addFileXML(const char *lfn,const StringBuffer &xml,IUserDescriptor *user)
+    void addFileXML(const char *lfn, const StringBuffer &xml, IUserDescriptor *user)
     {
         Owned<IPropertyTree> t = createPTreeFromXMLString(xml);
-        Owned<IFileDescriptor> fdesc = deserializeFileDescriptorTree(t,&queryNamedGroupStore(),0);
-        Owned<IDistributedFile> file = queryDistributedFileDirectory().createNew(fdesc,true);
-        if (file)
-            file->attach(lfn,user);
+        IDistributedFileDirectory &dfd = queryDistributedFileDirectory();
+        // Check if this XML is a superfile map
+        if (t->hasProp("@numsubfiles"))
+        {
+            if (dfd.exists(lfn, user))
+                throw MakeStringException(-1, "Super file '%s' already exists!", lfn);
+
+            // It seems XML is fine to create  super file
+            Owned<IDistributedSuperFile> superfile = dfd.createNewSuperFile(t);
+
+            superfile->validate();
+
+            try
+            {
+                superfile->attach(lfn, user, true);
```

Comment:
This is still an issue afaics.
The original comment(s) still stand, a few reviews ago still stands:
```
2) You do not want the init to loadSubFiles, because that will lookup and lock them and you are creating an unattached file (no Dali connection at this point). It should delay that action until attach().
```

I see there are a 2 critical typo's in my last comment above:

>but it should have loaded them

Should have been:
>but it **shouldn't** have loaded them.

>createNewSuperFile() should avoid loading the supers.

Should have been:
>createNewSuperFile() should avoid loading the **subfiles.**


**You do not want an attached superfile holding locks..**